lib/api/src/sys/imports.rs

```diff
@@ -130,16 +130,21 @@ impl Imports {
                 VMSharedMemory::new(&ty, &style).unwrap()
             });
 
+        let mut ret = self.clone();
         if let Some(memory) = shared_memory {
-            self.define(
-                "env",
-                "memory",
-                Memory::new_from_existing(store, memory.clone().into()),
-            );
-            Some(memory)
-        } else {
-            None
-        }
+            // if the memory has already be defined, don't redefine it!
+            if !self
+                .map
+                .contains_key(&("env".to_string(), "memory".to_string()))
+            {
+                ret.define(
+                    "env",
+                    "memory",
+                    Memory::new_from_existing(store, memory.clone().into()),
+                );
+            }
+        };
+        ret
```

Comment:
I still like to have this code outside of the API. It's veeeery tied to a specific need (and that's why it should be out of the API crate)

Reply:
Im' sory but I don't understand what you are asking. You want me to remove completly "import_shared_memory"?

Reply:
Yes. I believe we can achieve the same with a function method `into_shared` on the `Memory` export

Reply:
Ok, done